tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
+    this.promise = new Promise((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+  }
+  addEvent(eventData?: any) {
+    this.eventData = [
+      ...this.eventData,
+      eventData
+    ];
+    if (this.eventData.length >= this.expectedEvents) {
+      this.resolve(this.eventData);
+    }
+  }
+  reset(expectedEvents?: number) {
+    this.expectedEvents = expectedEvents || this.expectedEvents;
```

Comment:
Fair enough! We could refactor to add that sure. I only ever used the event counter but there were a couple cases where we could leverage the event matching if we wanted.

The tests pass as it stands, but they aren't as flexible as allowing for a condition might be.